common/ferrostar/src/routing_adapters/error.rs

```diff
@@ -53,7 +53,7 @@ impl From<serde_json::Error> for RoutingRequestGenerationError {
 #[derive(Debug)]
 #[cfg_attr(feature = "std", derive(thiserror::Error))]
 #[cfg_attr(feature = "uniffi", derive(uniffi::Error))]
-pub enum RoutingResponseParseError {
+pub enum OsrmParsingError {
```

Comment:
Is something about this specific to OSRM? Perhaps I'm too optimistic, but I think we can have a single error enum that covers what devs care about for most backends.